src/main/java/teammates/storage/api/FeedbackSessionsDb.java

```diff
@@ -544,6 +551,26 @@ public void deleteFeedbackSessionsForCourses(List<String> courseIds) {
     }
     
     @SuppressWarnings("unchecked")
+    private List<FeedbackSession> getFeedbackSessionEntitiesNeedingClosingEmail() {
+        Query q = getPm().newQuery(FeedbackSession.class);
+        q.declareParameters("boolean sentParam, boolean enableParam, Enum notTypeParam");
+        q.setFilter("sentClosingEmail == sentParam && isClosingEmailEnabled == enableParam "
+                    + "&& feedbackSessionType != notTypeParam");
+        
+        return (List<FeedbackSession>) q.execute(false, true, FeedbackSessionType.PRIVATE);
+    }
+    
+    @SuppressWarnings("unchecked")
+    private List<FeedbackSession> getFeedbackSessionEntitiesNeedingClosedEmail() {
+        Query q = getPm().newQuery(FeedbackSession.class);
+        q.declareParameters("boolean sentParam, boolean enableParam, Enum notTypeParam");
+        q.setFilter("sentClosedEmail == sentParam && isClosedEmailEnabled == enableParam "
+                    + "&& feedbackSessionType != notTypeParam");
```

Comment:
We can ignore the timezone. Just get the sessions having closing times within last 48 hours. This should bring down the result set to less than a hundred instead of 13k.